/*
 * Copyright (c) 2020 The Cyprestore Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file. See the AUTHORS file for names of contributors.
 */

syntax = "proto2";

import "common/pb/types.proto";

package cyprestore.extentmanager.pb;

option cc_generic_services = true;

message QueryRouterRequest {
    required string extent_id = 1;
}

message QueryRouterResponse {
    required cyprestore.common.pb.Status status = 1;
    optional uint32 router_version = 2;
    optional cyprestore.common.pb.ExtentRouter router = 3;
}

service RouterService {
  // router apis
  rpc QueryRouter(QueryRouterRequest) returns (QueryRouterResponse);
};
